pkg/stream/stream_start.go

```diff
@@ -50,6 +49,7 @@ func Start(ctx context.Context, logger loglib.Logger, config *Config, meter metr
 		if err != nil {
 			return fmt.Errorf("error setting up postgres replication handler")
 		}
+		defer replicationHandler.Close()
```

Comment:
can we really close this at the end of start? Would it mean `Run` is a better name?

Reply:
`Start` is a blocking call, when it exits it cleans up all started resources. That's where all others are defered closed/stopped. I can change it to `Run` if that's clearer though 👍 .